inc/database.class.php

```diff
@@ -0,0 +1,415 @@
+         'name'               => _n('Category', 'Categories', 1),
+         'datatype'           => 'dropdown'
+      ];
+
+      $tab[] = [
+         'id'                 => '41',
+         'table'              => State::getTable(),
+         'field'              => 'name',
+         'name'               => _n('State', 'States', 1),
+         'datatype'           => 'dropdown'
+      ];
+
+      $tab[] = [
+         'id'                 => '16',
+         'table'              => $this->getTable(),
+         'field'              => 'comment',
+         'name'               => __('Comments'),
+         'datatype'           => 'text'
+      ];
+
+      $tab[] = [
+         'id'                 => '19',
+         'table'              => $this->getTable(),
+         'field'              => 'date_mod',
+         'name'               => __('Last update'),
+         'datatype'           => 'datetime',
+         'massiveaction'      => false
+      ];
+
+      $tab[] = [
+         'id'                 => '121',
+         'table'              => $this->getTable(),
+         'field'              => 'date_creation',
+         'name'               => __('Creation date'),
+         'datatype'           => 'datetime',
+         'massiveaction'      => false
+      ];
+
+      $tab[] = [
+         'id'                 => '23',
+         'table'              => Manufacturer::getTable(),
+         'field'              => 'name',
+         'name'               => Manufacturer::getTypeName(1),
+         'datatype'           => 'dropdown'
+      ];
+
+      $tab[] = [
+         'id'                 => '24',
+         'table'              => User::getTable(),
+         'field'              => 'name',
+         'linkfield'          => 'users_id_tech',
+         'name'               => __('Technician in charge'),
+         'datatype'           => 'dropdown',
+         'right'              => 'own_ticket'
+      ];
+
+      $tab[] = [
+         'id'                 => '49',
+         'table'              => Group::getTable(),
+         'field'              => 'completename',
+         'linkfield'          => 'groups_id_tech',
+         'name'               => __('Group in charge'),
+         'condition'          => ['is_assign' => 1],
+         'datatype'           => 'dropdown'
+      ];
+
+      $tab = array_merge($tab, DatabaseInstance::rawSearchOptionsToAdd());
+      $tab = array_merge($tab, Notepad::rawSearchOptionsToAdd());
+
+      return $tab;
+   }
+
+
+   function post_addItem() {
+      //create default instance
+      $this->getFromDB($this->fields['id']);
+      $instance = new DatabaseInstance();
+
+      if ((!isset($this->input['instance_name']) || empty($this->input['instance_name'])) && isset($this->input['name'])) {
+         $this->input['instance_name'] = sprintf(
+            //TRANS: first parameter is the database name
+            __('"%1$s" default instance'),
+            $this->input['name']
+         );
+      }
+
+      $instance->add([
```

Comment:
I wonder if adding a default instance when all values are empty is really usefull.

Reply:
Well, it's been discussed; and the point was there is always an instance with a database; whether we got its information or not.
The API should support adding instances information; and there are a few information (like port) that can be specified when creating a database from the UI.